meson.build

```diff
@@ -0,0 +1,153 @@
+project(
+  'microshell',
+  'c',
+  version: '0.1.0'
+)
+
+project_description = 'Lightweight pure C implementation of virtual shell, compatible with VT100 terminal.'
+
+project_headers = [
+  'microshell.h',
+  'inc/ush.h',
+  'inc/ush_assert.h',
+  'inc/ush_commands.h',
+  'inc/ush_const.h',
+  'inc/ush_file.h',
+  'inc/ush_internal.h',
+  'inc/ush_node.h',
+  'inc/ush_shell.h',
+  'inc/ush_types.h',
+  'inc/ush_utils.h',
+]
+
+project_source_files = [
+  'src/src/ush.c',
+  'src/src/ush_assert.c',
+  'src/src/ush_read.c',
+  'src/src/ush_read_utils.c',
+  'src/src/ush_read_char.c',
+  'src/src/ush_parse.c',
+  'src/src/ush_parse_char.c',
+  'src/src/ush_parse_utils.c',
+  'src/src/ush_write.c',
+  'src/src/ush_write_utils.c',
+  'src/src/ush_prompt.c',
+  'src/src/ush_reset.c',
+  'src/src/ush_file.c',
+  'src/src/ush_node.c',
+  'src/src/ush_node_utils.c',
+  'src/src/ush_node_mount.c',
+  'src/src/ush_utils.c',
+  'src/src/ush_commands.c',
+  'src/src/ush_process.c',
+  'src/src/ush_autocomp.c',
+  'src/src/ush_autocomp_utils.c',
+  'src/src/ush_autocomp_state.c',
+  'src/src/commands/ush_cmd.c',
+  'src/src/commands/ush_cmd_cd.c',
+  'src/src/commands/ush_cmd_help.c',
+  'src/src/commands/ush_cmd_ls.c',
+  'src/src/commands/ush_cmd_pwd.c',
+  'src/src/commands/ush_cmd_cat.c',
+  'src/src/commands/ush_cmd_xxd.c',
+  'src/src/commands/ush_cmd_echo.c'
+]
+
+project_dependencies = [
+]
+
+include_directories = [
+]
+
+public_headers = include_directories('src')
+
+build_args = [
+  '-Werror',
+  '-Wall',
+  '-Wextra',
+  '-pedantic',
+]
```

Comment:
You should add `default_options: ['warning_level=3', 'werror=true']` to the `project()` function. Meson has these warning flags built in (and allows the user to override werror when the inevitable routine GCC update adds a bunch of new style warnings).